### PR TITLE
Document and extend release process

### LINKED ISSUE #363: Document release process
Add a `RELEASE.md` file documenting the release process, akin to https://github.com/cilium/cilium/blob/master/RELEASE.md and https://github.com/cilium/hubble/blob/master/RELEASE.md.

### PR DESCRIPTION
**Pull request by tklauser**
Document the current release process and slightly extend it in two ways:

* Add (and keep updated) a `stable.txt` file which always points to the latest stable version of Cilium CLI.
* Create automated releases as draft, create a changelog as part of the release process and let the maintainers publish the release after updating it with the changelog.

See individual commits for details.

Fixes #363

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+When a tag is pushed, a GitHub Action job takes care of creating a new GitHub
+draft release, building artifacts and attaching them to the draft release.
+
+The release notes need to be manually added before manually publishing the
+release.
+
+## (OPTIONAL) Update `stable.txt` in the master branch
+
+Hubble's installation instruction in the Cilium documentation uses the version specified in
```
Comment (rolinh) [inline RELEASE.md:98]: s/Hubble/Cilium CLI/

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+`stable.txt` in the master branch. There are a couple of things to consider when deciding
+whether to update `stable.txt`. Let's say `stable.txt` is currently pointing to `v0.6.1`:
+
+- If this is a minor or patch release relative to the current stable version (i.e. `v0.7.0`
+  or `v0.6.2`), update `stable.txt` so that people start picking up the new features / bug
+  fixes included in this release.
+- If this is a patch release of a previous version (e.g. `v0.5.2`), don't update
+  `stable.txt`.
+- If this is a major release (e.g. `v1.0.0`), the installation instructions in older Cilium
```
Comment (rolinh) [inline RELEASE.md:107]: This is taken straight from Hubble's `RELEASE.md`. I don't think it applies to Cilium CLI.

**Review by rolinh — CHANGES_REQUESTED** [review]

**Review by rolinh — APPROVED** [review]

File: RELEASE.md (hunk 2 above)
Comment (tklauser) [reply RELEASE.md]: Indeed, thanks. I was a bit to quick to push the PR. Now folded the `stable.txt` update into a section above.

**Review by nbusseneau — APPROVED** [review]
LGTM.

Thoughts: I'm thinking we could automate almost everything in there. Full process could be:
- A workflow that needs to be manually triggered, passing in version parameters. When triggered:
  - Prepare release notes.
  - Update `stable.txt` and `VERSION`.
  - Push branch and create PR.
- Then, another workflow running on push to `master`:
  - Identify that a release PR just got merged (file names filtering? commit names? PR name?)
  - Tag using version from PR
  - Create GH release using release notes from PR

**Comment by tklauser** [comment]
Thanks @nbusseneau! I've created #435 based on your comment https://github.com/cilium/cilium-cli/pull/434#pullrequestreview-707419013 so we don't forget to address it.

**Review by michi-covalent — APPROVED** [review]

**Pull request merged**